Adds option for 'simplified' header

laurelfulford opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

This PR adds an option to display a more simplified version of the header. 

This version of the menu will not show the 'secondary' or 'social' menus; I've added another item to #96 to explore adding an off-screen menu space where we can show those if it's found to be necessary. 

A couple known issues:
* The code in header.php is getting pretty gnarly -- there's a lot of checking for the different header settings to reorder things that could be simplified. This PR already has a lot going on in it though, so I plan to tackle that in a future PR; however, if the current state makes it hard to review, let me know and I can take a swing at it! 
* The 'stand out' link styling for the tertiary menu is currently applied to the last menu item; this needs to be switched so it can be applied with a class, to any item. 
* There might be some spacing weirdness if this is tested before #118 is merged; the extra markup one of those functions adds makes for extra space. 

The following screenshots should capture what some of the setting combinations should look like:

**Just 'Simplify Header':**

![image](https://user-images.githubusercontent.com/177561/61750340-fd9b0380-ad59-11e9-90d8-5e45cb08726d.png)

**Simplify Header + Solid Background:** 

![image](https://user-images.githubusercontent.com/177561/61750375-173c4b00-ad5a-11e9-940b-36749a28324f.png)

**Simplify Header + Centre Logo:**

![image](https://user-images.githubusercontent.com/177561/61750415-3509b000-ad5a-11e9-8218-4380fa4a71c0.png)

**Simplify Header + Centre Logo + Solid Background:**

![image](https://user-images.githubusercontent.com/177561/61750451-4f438e00-ad5a-11e9-9579-dbd965cdaef0.png)

**Bonus: + 'Hide Site Tagline':**

If tested after #117 is merged, hiding the site tagline should place the primary navigation next to the site title/logo on the left, rather than the tertiary navigation on the right:

![image](https://user-images.githubusercontent.com/177561/61750572-9cbffb00-ad5a-11e9-8315-5ea8e5409710.png)

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Make sure you have a primary and tertiary menu set up.
3. Navigate to Customizer > Site Identity and check 'Simplify Header'. 
4. Make sure nothing generally looks terrible.
5. Navigate to Customizer > Site Identity and test the 'Simplify Header' option with both the 'Solid Background' and 'Center Logo' options independently, and all three together. 
6. If testing after #117 is merged, can also be tested with 'Show Site Tagline' unchecked, specifically when 'Center Logo' is unchecked. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

laurelfulford commented:
> When the Logo is centered the Tagline displays regardless of the state of Display Tagline. (Note since that checkbox came in recently I rebased locally to include it in my testing)

I can't recreate this, but I might have inadvertently fixed it in my own rebasing/conflict fixing. Can you confirm if it's still happening?

> When Center Logo is checked, is the Tertiary menu supposed to appear to the left? I assumed the bordered item was always supposed to be on the far right, but with Tertiary on the left it's in an odd place:
> Relatedly, applying the boxed style to the last item in the menu feels like it could be limiting. Presumably this style will be generally used for DONATE or SUBSCRIBE action buttons, but if a publication doesn't have this button they will still be locked in to styling the final item in the nav this way. Is there another way this could be handled?

Yeah, this isn't great -- I buried this in my original comments, but I think a better approach to what I've done with highlighting this button would be to add a special CSS class to use. WordPress already lets you apply a CSS class to an individual menu item, and this would make it much more flexible than throwing it on the first or last menu item. The downside is that it adds complication to the setup -- do you think that's an okay trade off? 

> With Center Logo checked are we requiring that all menus be populated? If I unset Primary or Tertiary I end up with an imbalanced header like this:

Shoot, I meant to add a note about that! I need to add some kind of fallback in that case -- I was thinking of possibly moving the search to that side, so flexbox magically just did its thing without over complicating the CSS. I can dig into this -- would you be okay with my handling it in a separate PR? 

> Overall, is simplified header a setting that we want available in all style packs? Originally I assumed this was also going to be rolled into the appropriate style packs, but I'm not necessarily advocating for that, just noting that was my first thought...

My thinking with all the header options (and this might be _over_ thinking) is that I wanted to leave them mix and match so they could be worked in with each publication's branding (the solid background and ability to centre a logo, which would look better with some logos over others), and they wouldn't end up being restrictive (you can't fit as many menu items in this simplified version as you can with the primary menu on the bottom). I wasn't sure about tying them to a style pack because I didn't want to force together a style preference and header needs. I'm totally open to discussing this more, though! I could see arguments for going the other way with this, too, to simplify both the header code and the choices -- I might be over-optioning here.

laurelfulford commented on inc/customizer.php:
```diff
@@ -134,6 +134,24 @@ function newspack_customize_register( $wp_customize ) {
+		array(
+			'type'        => 'checkbox',
+			'label'       => esc_html__( 'Simplify Header', 'newspack' ),
+			'description' => esc_html__( 'Displays header as a shorter, simplier version.', 'newspack' ),
```
I initially missed this in the feedback -- the typo is fixed in 550111f!

laurelfulford commented:
> With Center Logo checked are we requiring that all menus be populated? If I unset Primary or Tertiary I end up with an imbalanced header like this:

I added a check for this in 943e446, and if there is no tertiary menu, an empty div will be output instead so the flexbox continues to work.

jeffersonrabb commented:
> I can't recreate this, but I might have inadvertently fixed it in my own rebasing/conflict fixing. Can you confirm if it's still happening?

Whatever it was, it no longer is. Confirmed...

jeffersonrabb commented:
> The downside is that it adds complication to the setup -- do you think that's an okay trade off?

I do think this is an ok tradeoff. I think it's better than having a weird menu item boxed, which may or may not be the kind of thing that should be called out like that. Would there be a way this could be switched on by a Customizer setting, or configured in the Newspack setup wizard?

jeffersonrabb commented:
> My thinking with all the header options (and this might be over thinking) is that I wanted to leave them mix and match so they could be worked in with each publication's branding (the solid background and ability to centre a logo, which would look better with some logos over others), and they wouldn't end up being restrictive (you can't fit as many menu items in this simplified version as you can with the primary menu on the bottom). I wasn't sure about tying them to a style pack because I didn't want to force together a style preference and header needs. I'm totally open to discussing this more, though! I could see arguments for going the other way with this, too, to simplify both the header code and the choices -- I might be over-optioning here.

This makes sense.

laurelfulford commented:
> I do think this is an ok tradeoff. I think it's better than having a weird menu item boxed, which may or may not be the kind of thing that should be called out like that. Would there be a way this could be switched on by a Customizer setting, or configured in the Newspack setup wizard?

I think it comes down to how much control we think people would want. 

What I'm suggesting -- a CSS class -- can only be added once the menu item you want to add it to exists. So I don't think there's a nice way to do that as a Customization option/Wizard thing, but it would be possible to use that and optionally hide a first-or-last menu item.

I've removed it for now in 9488b55 and will experiment in another PR.

> Is it true that the primary difference the simplified header offers is that it's shorter? If true, I'm wondering if we should just say that, and call it "Short Header?"

I'm good with that! I went with "simplified" because it won't show a couple of the menus, but "Short Header" is more descriptive. Updated in a5a8e0f.

> Because #116 landed while this PR was still open, there's a new spacing issue in the Simplified Header with the search icon.

Good catch! I added space in 73d2fd4.

laurelfulford commented:
Thanks for reviewing this monster! 😄
